python/ray/data/grouped_dataset.py

```diff
@@ -317,7 +318,7 @@ def group_fn(batch):
                 boundaries = get_key_boundaries(block_accessor)
             else:
                 boundaries = [block_accessor.num_rows()]
-            builder = block_accessor.builder()
+            builder = DelegatingBlockBuilder()
```

Comment:
Could we also update the docstring of the map_groups to include this as an example?

Reply:
@ericl - sure, added one. Please check if the example makes sense - get first value per group.